File: codes/sequential/instant/InstantParticle_Mother.py
from instant.InstantParticle import InstantParticle

class InstantParticle_Mother():
    def __init__(self, my_angle):
        self.particle_children_list = []
        self.my_angle = my_angle
        self.win_num = 0

    def appendChildren(self, position, map_value):
        self.particle_children_list.append(InstantParticle(position, map_value))

    def removeChildren(self, idx):
        del self.particle_children_list[idx]

    def getAvgWeight(self):
        sum = 0
        for c in self.particle_children_list:
            sum += c.weight
        if len(self.particle_children_list) == 0:
            result = 0
        else:
            result = sum / len(self.particle_children_list)
        return result

File: areasequential_11.21/mylibrary/src/main/java/com/example/mylibrary/instant/InstantParticle_Mother.kt
package com.example.mylibrary.instant

internal class InstantParticle_Mother constructor(angle : Int) {
    var particle_children_list = listOf<InstantParticle>()
    var my_angle : Int = 0
    var win_num : Int = 0
    init{
        my_angle = angle
    }
    fun appendChildren(position: ArrayList<Float>, map_value: Array<Double>){
        particle_children_list = particle_children_list.plus(InstantParticle(position, map_value))
    }
    fun removeChildren(idx: Int){
        particle_children_list = particle_children_list.minus(particle_children_list[idx])
    }

    // 210913 원준 수정
    fun getAvgWeight(): Double {
        var sum = 0.0
        var result = 0.0
        for (c in particle_children_list) {
            sum += c.weight
        }
        if (particle_children_list.size == 0){
            result = 0.0
        }
        else {
            result = sum / particle_children_list.size
        }
        return result
    }

}